### PR TITLE
Added license to all python files

### LINKED ISSUE #24: Add license as header to all source code files
Add the following license header to all Python files in the repo:

```py
# Copyright 2022 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
# SPDX-License-Identifier: Apache-2.0
```

as suggested in the [Copyright Header](https://github.com/ACCESS-NRI/template?tab=readme-ov-file#copyright-header) section of the ACCESS-NRI template repo.

### PR DESCRIPTION
**Pull request by atteggiani**
Fixes #24

### CODE REVIEW

File: amami/amami.py
Comment (truth-quark) [inline amami/amami.py:1]: This won't show up as the source file is removed in #20.

File: amami/misc_utils/__init__.py
Comment (truth-quark) [inline amami/misc_utils/__init__.py:1]: This edit will vanish in #20 due to renaming to `helpers.py`.

**Review by truth-quark — COMMENTED** [review]
I suggest deferring this PR until #20 is merged.

**Comment by atteggiani** [comment]
> I suggest deferring this PR until #20 is merged.

Yeah I thought about what would happen since there are modifications in those files. 
But if we merge this PR first, then won't _git_ handle it for the other one? 
We might get asked to solve solve some conflicts but wouldn't that be ok?

**Comment by truth-quark** [comment]
> > I suggest deferring this PR until #20 is merged.
> 
> Yeah I thought about what would happen since there are modifications in those files. But if we merge this PR first, then won't _git_ handle it for the other one? We might get asked to solve solve some conflicts but wouldn't that be ok?

I haven't come across this condition before. There might be a conflict with #33, even if this is merged 1st. It depends if git detects there's whitespace differences between the #34 version & the file deleted in PR #33 & flags a diff.

**Review by truth-quark — APPROVED** [review]
Good to go!

**Pull request merged**